app/models/github_user.rb

```diff
@@ -11,6 +11,11 @@ def accept_repository_invitation(invitation_id, **options)
     end
   end
 
+  def watch_repository(repo)
+    options = { subscribed: true }
+    @client.update_subscription(repo.github_repository.full_name, options)
```

Comment:
Methods in Octokit can throw a ton of different errors from the GitHub API, so in our GitHubResource classes, we usually wrap client calls. in `GitHub::Errors.with_error_handling do ... end`. The error handling manages all the different errors and raises `GitHub::Error` if something comes up. 

We should catch the error here and probably rescue and ignore the error, since it doesn't actually matter very much if we fail to follow the repo.